src/Pagination/Paginator.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace SumoCoders\FrameworkCoreBundle\Pagination;
+
+use Doctrine\ORM\QueryBuilder as DoctrineQueryBuilder;
+use Doctrine\ORM\Tools\Pagination\CountWalker;
+use Doctrine\ORM\Tools\Pagination\Paginator as DoctrinePaginator;
+use Traversable;
+
+class Paginator implements \Countable, \IteratorAggregate
+{
+    public const PAGE_SIZE = 10;
+
+    private DoctrineQueryBuilder $queryBuilder;
+    private int $currentPage;
+    private int $startPage;
+    private int $endPage;
+    private int $pageSize;
+    private Traversable $results;
+    private int $numResults;
+
+    public function __construct(DoctrineQueryBuilder $queryBuilder, int $pageSize = self::PAGE_SIZE)
+    {
+        $this->queryBuilder = $queryBuilder;
+        $this->pageSize = $pageSize;
+    }
+
+    public function paginate(int $page = 1): self
+    {
+        $this->currentPage = max(1, $page);
```

Comment:
Not a real issue, but if I pass 0 through the parameters it will show page 1. I would expect an exception or not found. But yours is more user-friendly.